Geovalidation.md

```diff
@@ -0,0 +1,143 @@
+| territories_as_states | Optional. When enabled, dependent territories may be requested as administrative areas of their parent nation, and will also be returned as such. This parameter is necessary to validate, for instance, **locality=San Juan&admin_area=PR&country=US**. This functionality will only recognize territories and parent countries by their [two-digit ISO-3166-1 country codes](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2), so requests such as **admin_area=Puerto Rico&country=USA** would not be remapped according to this parameter. Default value is false. |
+| try_locality_as_admin_area | Optional. When enabled, a request that includes the **locality** parameter and fails to retrieve locality-specific geography data will be reattempted with the locality value sent as an admin_area value instead. (If an admin_area value already exists, the locality value will be prepended and comma-separated, e.g. "locality=Harris%20Township&admin_area=OH" would be resent as "admin_area=Harris Township, OH" if try_locality_as_admin_area is set to true.) Default value is false. |
+| validationStrategies | Required. [Validation Strategies](#Validation Strategies) for making the geocoding request. |
+| acceptedLocationTypes | Required. [Location types](#Location types) to validate the geocoding response by. |
+
+&nbsp;
+#Validation Strategies
+
+| Validation Strategy | Description|
+|----------|-------------|
+| FIELDED | Make a Geocoder call with admin area, address, country, locality, and postal code |
+| FIELDED_NO_ADDRESS | Make a Geocoder call with the provided admin area, country, locality, and postal code |
+| FREETEXT | Makes a Geocoder call with the query parameter equal to the formated string "address, locality, admin area, country, postal code"  |
+| POSTAL_CODE_ONLY |  Makes a Geocoder call with postal code parameter set to the provided postal code |
+&nbsp;
+
+&nbsp;
+#Location Types
+
+| Location Types |
+|----------|
+| ADMIN_AREA1 |
+| ADMIN_AREA2 |
+| COORDINATE |
+| COUNTRY |
+| LOCALITY |
+| POSTAL_CODE |
+| STREET |
+| STREET_ADDRESS |
+| SUBLOCALITY |
+| UNKNOWN |
+
+&nbsp;
+
+&nbsp;
+
+----------------------
+#Response Structure
+
+All responses with an HTTP status of 200 will consist of a JSON object with a top-level "data" node containing the following elements:
+
+| Field    | Description |
+|----------|-------------|
+| strategy_results  | A JSON array of one or more objects containing the results of the requested strategies. The strategy_results will be ordered by their order in the validation_strategies comma separated string (ie for the string "FREETEXT,FIELDED" the result of FREETEXT will be the first result in the array) |
```

Comment:
so for someone coding against this API, what is the intended usage for retrieving a given strategy result? is it just taking results[0], results[1], etc? That doesn't feel very good to me. If we enforced ALL_CAPS casing in the request, then we could guarantee that the returned strategies would have the exact same names as the input strategies. Maybe our wrappers should expose a Response.GetResult(Strategy strategy) method to make the response easier to work with?